### ISSUE TITLE
panic: failed to parse CIDR 2804>3078::/32: invalid CIDR address

### ISSUE THREAD

**Issue by bmwiedemann** [ipfs/go-ipfs#8766]
### Checklist

- [X] This is a bug report, not a question. Ask questions on [discuss.ipfs.io](https://discuss.ipfs.io).
- [X] I have searched on the [issue tracker](https://github.com/ipfs/go-ipfs/issues?q=is%3Aissue) for my bug.
- [X] I am running the latest [go-ipfs version](https://dist.ipfs.io/#go-ipfs) or have an issue updating.

### Installation method

built from source

### Version

```Text
go-ipfs version: 0.12.0
Repo version: 12
System version: amd64/linux
Golang version: go1.16.14
```

### Config

```json
{
  "API": {
    "HTTPHeaders": {}
  },
  "Addresses": {
    "API": "/ip4/127.0.0.1/tcp/5001",
    "Announce": [
      "/ip4/188.40.142.18/tcp/34001/p2p/QmXxk7UJHRtiqD9JDbo3xgHTDdDPGPM9Q63PbGfCkShkXR",
      "/ip6/2a01:4f8:221:b52::2/tcp/34001/p2p/QmXxk7UJHRtiqD9JDbo3xgHTDdDPGPM9Q63PbGfCkShkXR"
    ],
    "Gateway": "/ip4/127.0.0.1/tcp/8080",
    "NoAnnounce": [
      "/ip4/10.0.0.0/ipcidr/8",
      "/ip4/169.254.0.0/ipcidr/16",
      "/ip4/192.168.0.0/ipcidr/16",
      "/ip4/172.16.0.0/ipcidr/12",
      "/ip4/240.0.0.0/ipcidr/4",
      "/ip6/2001:db8::/ipcidr/32",
      "/ip6/fc00::/ipcidr/7",
      "/ip6/fe80::/ipcidr/10"
    ],
    "Swarm": [
      "/ip4/0.0.0.0/tcp/34001",
      "/ip6/::/tcp/34001",
      "/ip4/0.0.0.0/udp/34001/quic",
      "/ip6/::/udp/34001/quic"
    ]
  },
  "Bootstrap": [
    "/dnsaddr/bootstrap.libp2p.io/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN",
    "/dnsaddr/bootstrap.libp2p.io/p2p/QmQCU2EcMqAqQPR2i9bChDtGNJchTbq5TbXJJ16u19uLTa",
    "/dnsaddr/bootstrap.libp2p.io/p2p/QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb",
    "/dnsaddr/bootstrap.libp2p.io/p2p/QmcZf59bWwK5XFi76CZX8cbJ4BhTzzA3gU1ZjYZcYW3dwt",
    "/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
    "/ip4/104.236.179.241/tcp/4001/p2p/QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
    "/ip4/128.199.219.111/tcp/4001/p2p/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu",
    "/ip4/104.236.76.40/tcp/4001/p2p/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64",
    "/ip4/178.62.158.247/tcp/4001/p2p/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd",
    "/ip6/2604:a880:1:20::203:d001/tcp/4001/p2p/QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
    "/ip6/2400:6180:0:d0::151:6001/tcp/4001/p2p/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu",
    "/ip6/2604:a880:800:10::4a:5001/tcp/4001/p2p/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64",
    "/ip6/2a03:b0c0:0:1010::23:1001/tcp/4001/p2p/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd",
    "/ip4/104.131.131.82/udp/4001/quic/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
  ],
  "Datastore": {
    "BloomFilterSize": 0,
    "GCPeriod": "10h",
    "HashOnRead": false,
    "Spec": {
      "mounts": [
        {
          "child": {
            "path": "blocks",
            "shardFunc": "/repo/flatfs/shard/v1/next-to-last/2",
            "sync": true,
            "type": "flatfs"
          },
          "mountpoint": "/blocks",
          "prefix": "flatfs.datastore",
          "type": "measure"
        },
        {
          "child": {
            "compression": "none",
            "path": "datastore",
            "type": "levelds"
          },
          "mountpoint": "/",
          "prefix": "leveldb.datastore",
          "type": "measure"
        }
      ],
      "type": "mount"
    },
    "StorageGCWatermark": 90,
    "StorageMax": "3000GB"
  },
  "Discovery": {
    "MDNS": {
      "Enabled": false,
      "Interval": 10
    }
  },
  "Experimental": {
    "FilestoreEnabled": true,
    "Libp2pStreamMounting": false,
    "P2pHttpProxy": false,
    "PreferTLS": false,
    "QUIC": false,
    "ShardingEnabled": false,
    "UrlstoreEnabled": false
  },
  "Gateway": {
    "APICommands": [],
    "HTTPHeaders": {
      "Access-Control-Allow-Headers": [
        "X-Requested-With",
        "Range",
        "User-Agent"
      ],
      "Access-Control-Allow-Methods": [
        "GET"
      ],
      "Access-Control-Allow-Origin": [
        "*"
      ]
    },
    "NoFetch": false,
    "PathPrefixes": [],
    "RootRedirect": "",
    "Writable": false
  },
  "Identity": {
    "PeerID": "QmXxk7UJHRtiqD9JDbo3xgHTDdDPGPM9Q63PbGfCkShkXR"
  },
  "Ipns": {
    "RecordLifetime": "",
    "RepublishPeriod": "",
    "ResolveCacheSize": 128
  },
  "Mounts": {
    "FuseAllowOther": true,
    "IPFS": "/ipfs",
    "IPNS": "/ipns"
  },
  "Pubsub": {
    "DisableSigning": false,
    "Router": "",
    "StrictSignatureVerification": false
  },
  "Reprovider": {
    "Interval": "12h",
    "Strategy": "all"
  },
  "Routing": {
    "Type": "dht"
  },
  "Swarm": {
    "AddrFilters": [
      "/ip4/10.0.0.0/ipcidr/8",
      "/ip4/169.254.0.0/ipcidr/16",
      "/ip4/192.168.0.0/ipcidr/16",
      "/ip4/172.16.0.0/ipcidr/12",
      "/ip4/240.0.0.0/ipcidr/4",
      "/ip6/2001:db8::/ipcidr/32",
      "/ip6/fc00::/ipcidr/7",
      "/ip6/fe80::/ipcidr/10"
    ],
    "ConnMgr": {
      "GracePeriod": "20s",
      "HighWater": 900,
      "LowWater": 600,
      "Type": "basic"
    },
    "DisableBandwidthMetrics": false,
    "DisableNatPortMap": true,
    "DisableRelay": false,
    "EnableAutoNATService": false,
    "EnableAutoRelay": false,
    "EnableRelayHop": false
  }
}
```

### Description

IPFS daemon fails to start, after it was working for some days with 0.12.0 (and with previous versions since 2019)

```
> ipfs daemon
Initializing daemon...
go-ipfs version: 0.12.0
Repo version: 12
System version: amd64/linux
Golang version: go1.16.14
panic: failed to parse CIDR 2804>3078::/32: invalid CIDR address: 2804>3078::/32

goroutine 23 [running]:
github.com/libp2p/go-libp2p-asn-util.newIndirectAsnStore.func1(0xc00007c000)
        pkg/mod/github.com/libp2p/go-libp2p-asn-util@v0.1.0/asn.go:91 +0xbe
created by github.com/libp2p/go-libp2p-asn-util.newIndirectAsnStore
        pkg/mod/github.com/libp2p/go-libp2p-asn-util@v0.1.0/asn.go:87 +0x8d
```

This uses the build from https://build.opensuse.org/package/show/filesystems/go-ipfs

**Comment by aschmahmann** [comment]
Does this happen reliably for you, or just occasionally? Have you tried rebuilding the binary to make sure you don't have some corruption locally or in your build process?

`panic: failed to parse CIDR 2804>3078::/32: invalid CIDR address: 2804>3078::/32` is a weird error to get since the error comes from https://cs.opensource.google/go/go/+/refs/tags/go1.16.14:src/net/ip.go;l=712 and returns the exact text that was passed in which comes from parsing the keys of a static [map](https://github.com/libp2p/go-libp2p-asn-util/blob/v0.1.0/ipv6_asn_map.gen.go) that is built into the binary, which while it contains `"2804:3078::/32"` does not contain the `>` character at all.

**Comment by bmwiedemann** [comment]
Thanks. It worked with the official binary and then I found that my binary on disk was not the one it should be. Exactly one `:` was replaced by `>` . Weird. And sure enough after a force-reinstall, it worked again.

smartctl show some HDD error
198 Offline_Uncorrectable   0x0008   100   100   000    Old_age   Offline      -       2

and the characters are just 1 bit apart, so let's close this here. Not a fault of IPFS, but of my machine corrupting the binary.
